### PR TITLE
Updating SingleListGrader to detect missing entries

### LINKED ISSUE #167: SingleListGrader and empty entries
If a list is entered as `1, 2, , 3`, this should probably generate an error due to a missing entry. Similarly for `1, 2, 3,`.

### PR DESCRIPTION
**Pull request by jolyonb**
Adds an option to SingleListGrader to detect and present an error if there are empty entries in a list, turned on by default.

Includes tests and documentation (I started converting the examples to doctests).

Resolves #167.

### CODE REVIEW

File: mitxgraders/listgrader.py [hunk 1]
```diff
@@ -692,6 +696,20 @@ def check_response(self, answer, student_input, **kwargs):
         answers = answer  # Rename from the ItemGrader name
         student_list = student_input.split(self.config['delimiter'])
 
+        # Check for empty entries in the list
+        if self.config['missing_error']:
+            bad_items = []
```
Comment (ChristopherChudzicki) [inline mitxgraders/listgrader.py]: `bad_items = [idx+1 for (idx, item) in enumerate(student_list) if item.strip() == '']` if you want.

File: docs/grading_lists/single_list_grader.md [hunk 2]
```diff
@@ -2,18 +2,26 @@
...
+True
+
 ```
 
 To receive full points for this problem, a student would enter `cat, dog` or `dog, cat` into the input box. Entering `cat, octopus` or just `cat` will receive half points. The `answers` key is a python list of the individual answers you wish to use.
 
 You can use a tuple of lists to specify multiple lists of answers, just like normal ItemGraders.
 
-```python
+```pycon
```
Comment (ChristopherChudzicki) [inline docs/grading_lists/single_list_grader.md]: I see change python -> pycon, but you still need the prompts `>>>` and `...` in order to make this run as a doctest. Also, I suspect the `pycon` syntax highlighting will look wrong without the prompts, but I am not sure.

(Oops, I guess I never submitted this comment in earlier review)

**Comment by jolyonb** [comment]
I think the problem is worst in SingleListGrader, because it's too easy to have an errant comma. In a ListGrader scenario, it's possible to leave a box empty, but I'm much less worried about that (sometimes it's by design).

File: mitxgraders/listgrader.py (hunk 1 above)
Comment (jolyonb) [reply mitxgraders/listgrader.py]: Yeah, that is neater.

File: docs/grading_lists/single_list_grader.md (hunk 2 above)
Comment (jolyonb) [reply docs/grading_lists/single_list_grader.md]: I think I was ambitious and thought I'd update the entire doc, but never quite got around to it... Well, maybe I'll just do so now.

**Comment by jolyonb** [comment]
Implemented your suggestion and updated the documentation properly. Ready for a final review.

**Pull request merged**